src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -569,6 +569,122 @@ void CodeGen::genHWIntrinsic_R_R_RM_I(GenTreeHWIntrinsic* node, instruction ins)
     }
 }
 
+//------------------------------------------------------------------------
+// genHWIntrinsic_R_R_R_RM: Generates the code for a hardware intrinsic node that takes two register operands,
+//                          a register/memory operand, and that returns a value in register
+//
+// Arguments:
+//    ins       - The instruction being generated
+//    attr      - The emit attribute
+//    targetReg - The target register
+//    op1Reg    - The register of the first operand
+//    op2Reg    - The register of the second operand
+//    op3       - The third operand
+//
+void CodeGen::genHWIntrinsic_R_R_R_RM(
+    instruction ins, emitAttr attr, regNumber targetReg, regNumber op1Reg, regNumber op2Reg, GenTree* op3)
+{
+    assert(targetReg != REG_NA);
+    assert(op1Reg != REG_NA);
+    assert(op2Reg != REG_NA);
+
+    emitter* emit = getEmitter();
+
+    if (op3->isContained() || op3->isUsedFromSpillTemp())
+    {
+        TempDsc* tmpDsc = nullptr;
+        unsigned varNum = BAD_VAR_NUM;
+        unsigned offset = (unsigned)-1;
+
+        if (op3->isUsedFromSpillTemp())
+        {
+            assert(op3->IsRegOptional());
+
+            tmpDsc = getSpillTempDsc(op3);
+            varNum = tmpDsc->tdTempNum();
+            offset = 0;
+
+            compiler->tmpRlsTemp(tmpDsc);
```

Comment:
Another possible comment for future: if there isn't a method to do this (I couldn't find it off-hand), perhaps there should be (returning a varNum):
```
tmpDsc = getSpillTempDsc(op3);
varNum = tmpDsc->tdTempNum();
compiler->tmpRlsTemp(tmpDsc);
```

Reply:
Added a `TODO-XArch-Cleanup` comment